### PR TITLE
Update 03 erc20 enforcer

### PR DESCRIPTION
**Pull request by mmvazzano**
Add the missing tests in 03_Erc20Enforcer.t.sol
Add the PermitSigUtils contract needed for the test that validates a signature

### CODE REVIEW

File: test/foundry/utils/PermitSigUtils.sol [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+pragma solidity 0.8.13;
+
+contract PermitSigUtils {
+  bytes32 internal DOMAIN_SEPARATOR;
+
+  constructor(bytes32 _DOMAIN_SEPARATOR) {
+    DOMAIN_SEPARATOR = _DOMAIN_SEPARATOR;
+  }
+
+  // keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
```
Comment (fgregoribattista) [inline test/foundry/utils/PermitSigUtils.sol:11]: Does this comment refer to the byte representation of the following line?

**Review by fgregoribattista — COMMENTED** [review]
Very good work, for me it's ready to be reviewed.

File: test/foundry/utils/PermitSigUtils.sol (hunk 1 above)
Comment (mmvazzano) [reply test/foundry/utils/PermitSigUtils.sol:11]: Yes, that is correct

**Review by sogoiii — APPROVED** [review]
👍

**Comment by mmvazzano** [comment]
@sogoiii I found a bug in the interface testing and modified the function testSupportinterface()

**Review by alexanderattar — APPROVED** [review]

**Pull request merged**